sdp-utils.c

```diff
@@ -906,31 +909,58 @@ const char *janus_sdp_get_fmtp(janus_sdp *sdp, int pt) {
 	return NULL;
 }
 
+/*! \brief Concatenates a string to another and dynamically increases the buffer if it is too small
+ * @param dynamic_buffer - Pointer to the dynamic buffer for the target string. This must be a dynamic allocated memory.
+ * @param add - String to add
+ * @param size - Current size of the buffer
+ * @returns the size of the buffer (may exceed size if it was resized)
+ */
+gsize dynamic_strlcat(char** dynamic_buffer, const char* add, gsize size) {
+	// What is the resulting length (current used buffer size + length of the addition + null byte)
+	size_t required = strlen(*dynamic_buffer) + strlen(add) + 1;
+	// In case the buffer is not large enough
+	if(required > size) {
+		if(required > MAX_JANUS_BUFSIZE)
+			JANUS_LOG(LOG_ERR, "dynamic_strlcat - Required buffer size %ld exceeds MAX_JANUS_BUFSIZE (%d)", required, MAX_JANUS_BUFSIZE);
+		// Either we double the size, or we increase by the required size
+		size_t newsize = MAX(size * 2, required);
+		if(newsize > MAX_JANUS_BUFSIZE)
+			newsize = MAX_JANUS_BUFSIZE;
+		if(newsize > size) {
+			size = newsize;
+			*dynamic_buffer = g_realloc(*dynamic_buffer, size);
+		}
+	}
+	g_strlcat(*dynamic_buffer, add, size);
```

Comment:
I'd still check what the return value of this function is, just in case there can be other causes of issues. Maybe we can do something like this instead:

    return janus_strlcat(*dynamic_buffer, add, size);

since that method has the necessary check in place, and so we can delegate it to that.